### PR TITLE
config: Add GoMemLimit config option and use with 10-node test

### PR DESCRIPTION
**Pull request by ohill**
## Summary

This adds a config option GoMemLimit for a soft memory limit like GOMEMLIMIT. It uses it for TestPartitionHalfOffline which runs 10 nodes (TenNodesDistributedMultiWallet.json) for 1GB each. It also uses it for TestSendSigsAfterCatchpointCatchup which starts 3 nodes and has one generate a catchpoint snapshot for the others.

## Test Plan

TestPartitionHalfOffline should run 10 nodes without using as much memory, and TestSendSigsAfterCatchpointCatchup should also use less memory.

### CODE REVIEW

File: test/e2e-go/features/catchup/stateproofsCatchup_test.go [hunk 1]
```diff
@@ -204,9 +204,9 @@ func TestSendSigsAfterCatchpointCatchup(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	defer fixtures.ShutdownSynchronizedTest(t)
 
-	if testing.Short() {
-		t.Skip()
-	}
+	// if testing.Short() {
```
Comment (gmalouf) [inline test/e2e-go/features/catchup/stateproofsCatchup_test.go:207]: why comment this out?

File: test/e2e-go/features/partitionRecovery/partitionRecovery_test.go [hunk 2]
```diff
@@ -231,9 +231,9 @@ func TestPartitionHalfOffline(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	defer fixtures.ShutdownSynchronizedTest(t)
 
-	if testing.Short() {
-		t.Skip()
-	}
+	//if testing.Short() {
```
Comment (gmalouf) [inline test/e2e-go/features/partitionRecovery/partitionRecovery_test.go:234]: same comment, not sure we should merge this as-is

File: test/e2e-go/features/partitionRecovery/partitionRecovery_test.go [hunk 3]
```diff
@@ -251,6 +251,7 @@ func TestPartitionHalfOffline(t *testing.T) {
 		a.NoError(err)
 		// adjust the refresh interval for one hour, so that we won't be reloading the participation key during this test.
 		cfg.ParticipationKeysRefreshInterval = time.Hour
+		cfg.GoMemLimit = 1 * 1024 * 1024 * 1024
 		cfg.SaveToDisk(nodeDir)
 	}
 	fixture.Start()
```
[the remark this answers is not in the document]
Comment (gmalouf) [reply test/e2e-go/features/partitionRecovery/partitionRecovery_test.go:257]: Feels heavy to add a new template variable for this

**Review by gmalouf — APPROVED** [review]
I think a config option is reasonable enough, understanding we may want to later remove if Golang changes the expected behavior/approach.

**Review by algorandskiy — COMMENTED** [review]
LGTM but lets merge after p2p branch since it also adds v34 config, and rebasing this smaller PR will be easier.

**Review by algorandskiy — APPROVED** [review]
Well although feature/p2p has a conflict so requires manual remerge anyway, lets get it merged.

**Pull request merged**